lib/puppet/functions/stdlib/ssl/validate_key_pair.rb

```diff
@@ -0,0 +1,27 @@
+require 'openssl'
+# @summary
+#    Validates a PEM-formatted X.509 public_key and private key using
+#    OpenSSL. Verifies that the public_key's matches the supplied key.
+Puppet::Functions.create_function(:'stdlib::ssl::validate_key_pair') do
+  dispatch :validate_ssl_key_pair do
+    param 'String[1]', :public_key
+    param 'String[1]', :private_key
+    return_type 'Boolean'
+  end
+  def validate_ssl_key_pair(public_key, private_key)
+    begin
+      private_key = OpenSSL::PKey.read(private_key)
+    rescue OpenSSL::PKey::PKeyError => error
+      raise Puppet::ParseError, "Not a valid Private key: #{error}"
```

Comment:
is it best to raise or just return false here?  or just let the original exception bubble up?

Reply:
added althugh now wonder if this needs to be split into two separate PR's :(